ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/util/PersistentQueryCleanupImpl.java

```diff
@@ -47,20 +48,23 @@ public void cleanupLeakedQueries(final List<PersistentQueryMetadata> persistentQ
         .stream()
         .map(PersistentQueryMetadata::getQueryApplicationId)
         .collect(Collectors.toSet());
-    try {
-      final Set<String> allStateStores = new HashSet<>(Arrays.asList(new File(stateDir).list()));
+
+    final String[] stateDirFileNames = new File(stateDir).list();
```

Comment:
Is there a change that `new File(stateDir).list()` would cause a NPE?

Reply:
No change. IntelliJ showed a warning that `list()` may return `null` (if `stateDir` is not a directory) and `asList()` would throw a NPE if you pass in `null`.